Add first cut of auth client

MrCreosote opened the pull request:
Doesn't do much yet but should have all the guts.

Back to standard ROE for this PR.

briehl reviewed (approved):
LGTM

Xiangs18 commented on src/main/java/us/kbase/auth/AuthException.java:
```diff
@@ -0,0 +1,28 @@
+	// TODO CODE could add the various fields from the auth server so that services can
+	//           react differently to different error codes
+
+	private static final long serialVersionUID = 1L;
```
Is this used anywhere?

MrCreosote replied:
it's for serialization: https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/io/Serializable.html

Xiangs18 replied:
👍

MrCreosote commented on src/main/java/us/kbase/auth/client/AuthClient.java:
```diff
@@ -0,0 +1,139 @@
+			final Map<String, Object> obj; 
+			try {
+				@SuppressWarnings("unchecked")
+				final Map<String, Object> foo = MAPPER.readValue(res, Map.class);
+				obj = foo;
```
You can't put the unchecked annotation directly on obj, it has to go on a variable declaration. the compiler will compile away `foo` in any case

Xiangs18 replied:
👍

Xiangs18 commented on build.gradle:
```diff
@@ -51,6 +51,13 @@ javadoc {
+	// using older dependencies to not force upgrades on services that might not be able to
+	// handle them. Need to upgrade the services and then upgrade here
+	implementation 'ch.qos.logback:logback-classic:1.1.2'
+	implementation 'com.fasterxml.jackson.core:jackson-annotations:2.5.4'
```
do we need annotations here? It seems only databind is used.

Xiangs18 replied:
👍

Xiangs18 reviewed (approved):
LGTM!

MrCreosote commented:
now if &*&()^&_ codecov would work

MrCreosote commented:
Ok, codecov isn't happy right now. The only change since the last report is removing the annotations jar so the report is still valid, so I'm merging.
